tests/src/test/scala/cats/collections/DietSpec.scala

```diff
@@ -1,375 +1,231 @@
-  test("diet")(forAll(arbNine) { r =>
-    val d = fromRanges(r)
-
-    (0 to 1000).toList.foreach { i =>
-      if (d.contains(i) != r.contains(i)) {
-        println(s"for $i, got ${d.contains(i)} expected ${r.contains(i)}")
-      }
+  implicit val arbItem: Arbitrary[Item] =
+    Arbitrary(
+      Gen.oneOf(
+        Gen.choose(0, 999).map(Item.Single),
+        for (min <- Gen.choose(0, 999); max <- Gen.choose(0, 999)) yield Item.Multiple(min, max)
+      )
+    )
+
+  case class Ranges(rs: List[(Boolean, Item)]) {
+    def isEmpty: Boolean = toSet.isEmpty
+    def toSet: Set[Int] = rs.foldLeft(Set.empty[Int]) { (set, r) =>
+      val s = r._2.toSet
+      if (r._1) set.union(s)
+      else set.diff(s)
     }
-    (0 to 1000).toList.forall(i => {
-      d.contains(i) == r.contains(i)
-    })
-    ()
-  })
-
-  test("min")(forAll(arbNine)(rs => rs.min ==  fromRanges(rs).min))
-  test("max")(forAll(arbNine)(rs => rs.max ==  fromRanges(rs).max))
-
-  test("merge")(forAll(arbNine, arbNine)  { (r1, r2) =>
-    val d = Diet.empty[Int] ++ fromRanges(r1) ++ fromRanges(r2)
-
-    (0 to 1000).toList.foreach {i =>
-      if(d.contains(i) != (r1.contains(i) || r2.contains(i)))
-        println(s"for $i, got ${d.contains(i)} expected ${r1.contains(i)} || ${r2.contains(i)}")
+    def toDiet: Diet[Int] = rs.foldLeft(Diet.empty[Int]) { (diet, r) =>
+      if (r._1) r._2.addToDiet(diet)
+      else r._2.removeFromDiet(diet)
     }
-
-    (0 to 1000).toList.forall(i => d.contains(i) == (r1.contains(i) || r2.contains(i) ))
-    ()
-  })
-
-
-//  property("match") = forAll{ (r: Ranges) =>
-//    val d = fromRanges(r)
-//
-//    val matcher = matchTo(d)
-//
-//    matcher.apply(d).matches == true
-//  }
-}
-/*
-class DietTest extends DogsSuite {
-  import Diet._
-
-  test("join nodes when item adj to existing seq"){
-    val diet = Diet.empty[Int].add(5).add(6).add(1).add(3).add(2).add(8)
-
-    val result = diet.intervals.map(l => l.generate)
-
-    result should matchTo(List[List[Int]](List(1, 2, 3), List(5, 6), List(8)))
   }
 
-  test("be always sorted"){
-    val diet = Diet.empty[Int].add(5).add(6).add(1).add(3).add(2).add(8)
+  implicit val arbRanges: Arbitrary[Ranges] =
+    Arbitrary(
+      Gen.listOf {
+        for {
+          inout <- Gen.oneOf(true, false)
+          item <- Arbitrary.arbitrary[Item]
+        } yield (inout, item)
+      }.map(Ranges)
+    )
 
-    val sorted = diet.toList()
+  implicit val arbDiet: Arbitrary[Diet[Int]] = Arbitrary(arbRanges.arbitrary.map(_.toDiet))
 
-    sorted should matchTo(List(1, 2, 3, 5, 6, 8))
-  }
+  // TODO requires a reasonable implementation!
+  implicit def dietEq: Eq[Diet[Int]] = Eq.instance((x, y) => x.toList == y.toList)
 
-  test("add disjoint range"){
+  test("shown empty"){
     val diet = Diet.empty[Int]
 
-    val result = diet.addRange(Range(0, 100))
-
-    val other = result.intervals.map(l => l.generate)
-
-    other should matchTo(List(Range(0, 100).toList))
+    diet.show should be ("{ }")
```

Comment:
Personally I’d like the string Diet to appear somewhere here

Reply:
This is a separate discussion, I think. I'll open a ticket.